analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,329 @@
+
+First, create genomic ranges objects for CNV seg file and gene.
+
+```{r}
+# Genomic range for cnv seg file
+cnv_gr <- cnv %>%
+  # format
+  dplyr::rename(
+    chr = chrom, start = loc.start, end = loc.end,
+    copy_number = copy.num
+  ) %>%
+  dplyr::select(-num.mark, -seg.mean) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# Genomic range for gene location and domain overlap file
+domain_gr <- bioMartDataPfamFGFR1 %>%
+  # keep only domain info
+  dplyr::filter(!is.na(domain_start), !is.na(domain_end)) %>%
+  # format
+  dplyr::mutate(
+    strand = if_else(strand == "-1", "-", "+"),
+    domain_chr = paste0("chr", domain_chr)
+  ) %>%
+  dplyr::rename(chr = domain_chr, start = domain_start, end = domain_end) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE,
+    seqnames.field = "chr"
+  )
+```
+
+
+
+### Distribution of manta SV
+SV should be annotated as tandem duplicate and overlap 
+"Protein tyrosine kinase" domain of FGFR1 in biospecimens
+
+```{r}
+
+# Genomic range for manta file
+manta_gr <- manta_lgat %>%
+  # format
+  dplyr::rename(
+    chr = SV.chrom, start = SV.start, end = SV.end,
+    ALT = ALT
+  ) %>%
+  # format with chr to match to cnv
+  dplyr::mutate(chr = paste0("chr", chr)) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# overlap manta and domain
+overlaps_manta <- IRanges::subsetByOverlaps(manta_gr, domain_gr)
+
+# filter for fgfr only overlap
+overlaps_manta_fgfr <- overlaps_manta[which(overlaps_manta$Gene.name == "FGFR1" &
+  # filter for tandem duplication
+  overlaps_manta$ALT == "<DUP:TANDEM>"), ]
+
+overlaps_manta_fgfr_df <- data.frame(
+  biospecimen_id = overlaps_manta_fgfr$Kids.First.Biospecimen.ID.Tumor,
+  ALT = overlaps_manta_fgfr$ALT,
+  # annnotation for dup tandem
+  FGFR_DUP_TANDEM = "Yes",
+  stringsAsFactors = FALSE
+)
+
+overlaps_manta_fgfr_df
+```
+4 biospecimens have tandem duplication that overlap tyrosine kinase domain in FGFR1
+
+
+### Distribution of copy number
+CNV that are within FGFR1 gene region only and overlap "Protein tyrosine kinase" domain in FGFR1 in biospecimens
+
+```{r}
+
+# gene region overlap
+fgfr_gene <- GenomicRanges::GRanges(
+  seqnames = paste0("chr", domain_gr$chromosome_name),
+  ranges = IRanges::IRanges(start = domain_gr$gene_start, end = domain_gr$gene_end)
+)
+
+# subset cnv+domain grange to be within FGFR1 gene region
+cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene, type = "within", maxgap = )
+
+# overlap consensus seg with exon
+overlaps_fgfr_cnv <- IRanges::subsetByOverlaps(cnv_fgfr_gr, domain_gr)
```

Comment:
So is this overlap showing whether or not the previous overlap contains an protein tyrosine kinase exon?

If yes, can we make the note more specific?

Reply:
Is there a reason we can't do this in one overlap? I think we should be able to do that. It doesn't look like you return to using `cnv_fgfr_gr` for anything.

Reply:
>So is this overlap showing whether or not the previous overlap contains an protein tyrosine kinase exon?
Right ^ I can update the comment to specify the same.

I  didn't quite figure out how to put it in 1 step such that we can find CNVS that overlaps domain_gr and are also fully within FGFR gene only (ie. CNV is smaller than FGFR gene body). 
I guess a maxgap of min[( domain_start - gene_start ), (gene_end - domain_end)] might work, checking ..

Reply:
I update the 2 step above to 1 step: 

```r
# subset cnv+domain grange to be within FGFR1 gene region 
cnv_fgfr_gr <- IRanges::subsetByOverlaps(
  cnv_gr,
  fgfr_gene,
  # within FGFR1 gene
  type = "within",
  # has atleast min overlap of domain region
  minoverlap = width(ranges(domain_gr)))

```

So this way the subset if within "FGFR1" gene body and has minOverlap domain_gr. We can argue that the minOverlap region could also be outside the actual domain_gr , but this is the only way i could think of making it 1 step. Let me know if you had any other way.

Reply:
> CNVS that overlaps domain_gr and are also fully within FGFR gene only

Does this mean you do not want to identify a CNV that is longer than the FGFR gene? Why is that? But yes you could should be able to use maxgap for that.

type="within" (based on my reading of the documentation) won't exclude things based on the CNV being longer than the FGFR gene either. So even if you keep both steps, I think you'd still need to specify a maxgap parameter.

Reply:
>Does this mean you do not want to identify a CNV that is longer than the FGFR gene? Why is that? But yes you could should be able to use maxgap for that.

CNVs need to be smaller that FGFR1 gene body, actually we only want to eventually retain CNVs that are duplication of exon10 to exon18 Or the kinase domain because duplication of this specific region defines this subtype.  

>type="within" (based on my reading of the documentation) won't exclude things based on the CNV being longer than the FGFR gene either. So even if you keep both steps, I think you'd still need to specify a maxgap parameter. 

Does a maxgap of gene body minus domain sound like a good maxgap value? It seems to give the same 5 bs ids as overlap

Reply:
Okay. Then yes, I think that plan makes sense.